package.json

```
@@ -24,25 +24,30 @@
   ],
   "files": [
     "index.js",
-    "regex.js"
+    "regex.js",
+    "*.d.ts"
   ],
-  "dependencies": {},
   "devDependencies": {
+    "@types/tape": "^4.0.0",
     "@unicode/unicode-13.0.0": "^1.0.0",
     "nyc": "^15.0.0",
     "prettier": "^2.0.0",
     "regenerate": "^1.0.0",
     "remark-cli": "^9.0.0",
     "remark-preset-wooorm": "^8.0.0",
+    "rimraf": "^3.0.0",
     "tape": "^5.0.0",
+    "typescript": "^4.0.0",
     "xo": "^0.36.0"
   },
   "scripts": {
     "generate": "node build",
     "format": "remark . -qfo && prettier . -w --loglevel warn && xo --fix",
     "test-api": "node test",
     "test-coverage": "nyc --reporter lcov tape test.js",
-    "test": "npm run generate && npm run format && npm run test-coverage"
+    "build": "rimraf \"*.d.ts\" && tsc",
+    "test": "npm run generate && npm run format && npm run test-coverage && npm run build",
+    "prepack": "npm run build"
```

```suggestion
    "prepack": "npm run build && npm run format"
```

I think it’d be nice to at least have prettier and such run over the generated types too

---

@ChristianMurphy and I were experimenting with `type-coverage` yesterday. It’s pretty useful. You might want try it out here too?

Here’s what I think I’m settling for: https://github.com/wooorm/zwitch/blob/480a9ea4446d7e6630e2e413ab2a768e2b8f529c/package.json#L43